addons/mail/static/src/discuss/call/common/rtc_service.js

```diff
@@ -474,6 +476,33 @@ export class Rtc extends Record {
         }, PING_INTERVAL);
     }
 
+    get displaySurface() {
+        return this.state.sourceScreenStream?.getVideoTracks()[0]?.getSettings().displaySurface;
+    }
+
+    showMirroringWarning() {
+        this.state.screenTrack.enabled = false;
+        this.removeMirroringWarning = this.overlay.add(
+            CallInfiniteMirroringWarning,
+            {
+                onClose: ({ stopScreensharing } = {}) => {
+                    this.removeMirroringWarning({ stopScreensharing });
+                },
+            },
+            {
+                onRemove: ({ stopScreensharing } = {}) => {
+                    if (stopScreensharing) {
+                        this.toggleVideo("screen", false);
+                    }
+                    this.removeMirroringWarning = null;
+                },
+            }
+        );
+        this.state.screenTrack.addEventListener("ended", () => this.removeMirroringWarning?.(), {
```

Comment:
the gain is probably too small to change that, but it's adding a new event listener everytime `showMirroringWarning` is called